client/errors.go

```diff
@@ -15,8 +15,11 @@ func ErrorClassifier(_ schema.ClientMeta, resourceName string, err error) diag.D
 func classifyError(err error, fallbackType diag.Type, opts ...diag.BaseErrorOption) diag.Diagnostics {
 	ie := errors.Unwrap(err)
 	if se, ok := ie.(k8s.APIStatus); ok {
-		if se.Status().Code == 403 {
+		switch se.Status().Code {
+		case 403:
 			return diag.FromError(ie, diag.ACCESS, diag.WithSeverity(diag.WARNING), diag.WithDetails(se.Status().Details.String()))
+		case 404:
+			return diag.FromError(ie, diag.RESOLVING, diag.WithSeverity(diag.IGNORE), diag.WithDetails("Current version of k8s might not support requested resource"))
```

Comment:
```suggestion
			return diag.FromError(ie, diag.RESOLVING, diag.WithSeverity(diag.IGNORE), diag.WithDetails("Current version of k8s might not support the requested resource"))
```

Can we recommend an action item to the user? e.g. updating something with their `k8s` environment?

Reply:
To follow up on my intention with my comment, does the following message makes sense:
```
"Current version of k8s might not support the requested resource. Consider upgrading your k8s to the most recent version"
```

Also, is this something we can detect in advance? If a k8s version doesn't support a resource, we shouldn't try to fetch it. Is this a valid action item to handle in a separate future PR?